Add executor and fuser options to the fastrnn test fixture

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42946 Add executor and fuser options to the fastrnn test fixture**

Summary: There are 3 options for the executor and fuser and some of them aren't
super interesting so I've combined the options into a single parameter, but
made it fairly easy to expand the set if there are other configs we might care
about.

Test Plan: Benchmark it

Reviewers: mvz, whc

Subscribers:

Tasks:

Tags:

Differential Revision: [D23090177](https://our.internmc.facebook.com/intern/diff/D23090177)

bertmaher committed:
Add executor and fuser options to the fastrnn test fixture

Summary: There are 3 options for the executor and fuser and some of them aren't
super interesting so I've combined the options into a single parameter, but
made it fairly easy to expand the set if there are other configs we might care
about.

Test Plan: Benchmark it

Reviewers: mvz, whc

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

wconstab commented:
this looks good, did you confirm that the fuser/executor setting sticks correctly throughout the scope of the forward/backward for a given test?  I should think it would..

also, i'd be interested to see how many minutes this adds to the benchmark, let's take a look when it finishes running

finally, not sure if we want to change the namespace of the tests with the legacy string.  (maybe we do, but if not, there is a workaround: supply 'ids' kwarg to parametrize giving a list of the same length as the actual parametrization but where the one corresponding to legacy is an empty string.  This would avoid a discontinuity in the named series already collected by scuba, but would be less explicit, and i'm not heartbroken if we reset the naming after only a week of data...

wconstab commented:
https://github.com/pytorch/hub/blob/master/benchmark/test_bench.py#L29
example of using 'ids' field in this case to display short names instead of the full file paths that are the actual parameter

bertmaher commented:
Ah yeah the scuba series is something I wanted your thoughts on.  I could add columns for the fuser and executor to the scribe request, which would be kind of nice for comparative analysis.  It would muddy the dataset if you forget to filter on those columns, though, but we could at least fix that in the dashboard.  What do you think?

bertmaher commented:
Another interesting finding, I think the default configuration corresponds to simple-none, which is kind of odd but hey.  Should we add that config to the parameters?

wconstab commented:
RE scuba series: i'm not sure which is better, new columns or not.  New columns kind of sounds 'right' to me, although we'd probably also want to clean up the RNNs test naming scheme, where 'jit' is a test name rather than a 'column' applied to the existing rnn test.

Another option to note, for making filtering easier, see [group=fastrnns](https://github.com/pytorch/pytorch/blob/master/benchmarks/fastrnns/test_bench.py#L37) which is mapped already to 'benchmark_group' in scribe.  If you wanted to leave the existing data alone but start working towards data with new sets of columns, you could make your version of the benchmark run with a different group tag somehow.

RE default configuration- I think you mean that all the current benchmarks are running under whatever the default pytorch settings are? I'm not sure why this was a revelation though, as you're aware of needing to add the new nnc-relevant configs?  OR are you saying that it's currently not even using the defaults, but force disabling everything?

bertmaher commented:
> this looks good, did you confirm that the fuser/executor setting sticks correctly throughout the scope of the forward/backward for a given test? I should think it would..

Good thought, I went and checked this locally by setting `PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser"` and verifying that the TE-fused IR is only dumped when the TE fuser is on.  It looks good!

ZolotukhinM commented:
> Another interesting finding, I think the default configuration corresponds to simple-none, which is kind of odd but hey. Should we add that config to the parameters?

FWIW, it was default in 1.5, but for 1.6 the default is 'legacy-old'.

bertmaher commented:
> > Another interesting finding, I think the default configuration corresponds to simple-none, which is kind of odd but hey. Should we add that config to the parameters?
> 
> FWIW, it was default in 1.5, but for 1.6 the default is 'legacy-old'.

Really?  When I don't set anything explicitly it looks like I'm getting the simple executor.  But maybe I'm off base here...

bertmaher committed:
Update on "Add executor and fuser options to the fastrnn test fixture"


Summary: There are 3 options for the executor and fuser and some of them aren't
super interesting so I've combined the options into a single parameter, but
made it fairly easy to expand the set if there are other configs we might care
about.

Test Plan: Benchmark it

Reviewers: mvz, whc

Subscribers:

Tasks:

Tags:

Differential Revision: [D23090177](https://our.internmc.facebook.com/intern/diff/D23090177)

[ghstack-poisoned]

bertmaher commented:
Here's a proposal for new columns.  We'll keep `benchmark_name` untouched (although I did some munging in the uploader script to keep the `test_direction[name]` formatting), and add `benchmark_executor` and `benchmark_fuser`.  Something I really like about using separate columns is that we can easily do a comparison view in scuba to see old vs te perf in a nice table.

ZolotukhinM commented:
> > > Another interesting finding, I think the default configuration corresponds to simple-none, which is kind of odd but hey. Should we add that config to the parameters?
> > 
> > 
> > FWIW, it was default in 1.5, but for 1.6 the default is 'legacy-old'.
> 
> Really? When I don't set anything explicitly it looks like I'm getting the simple executor. But maybe I'm off base here...

Oh, I should've been more specific. On master it was and still is 'simple-none', but if you look at the releases, then 1.6 would be 'legacy-old'. We decided to keep the master on 'simple-none' in a hope to switch to 'profiling-te' in the next release.

wconstab reviewed (approved):
LGTM!
